homeassistant/components/remote/harmony.py

```diff
@@ -196,101 +204,161 @@ def is_on(self):
         """Return False if PowerOff is the current activity, otherwise True."""
         return self._current_activity not in [None, 'PowerOff']
 
+    @property
+    def available(self):
+        return self._available
+
     async def async_update(self):
         """Retrieve current activity from Hub."""
         _LOGGER.debug("Updating Harmony.")
         if not self._client.config:
             await self._client.get_config()
 
-        activity_id = await self._client.get_current_activity()
-        activity_name = self._client.get_activity_name(activity_id)
+        activity_id, activity_name = self._client.current_activity
         _LOGGER.debug("%s activity reported as: %s", self._name, activity_name)
         self._current_activity = activity_name
         self._state = bool(self._current_activity != 'PowerOff')
         return
 
-    def new_activity(self, activity_id):
+    def new_activity(self, activity_info):
         """Call for updating the current activity."""
-        activity_name = self._client.get_activity_name(activity_id)
+        activity_id, activity_name = activity_info
         _LOGGER.debug("%s activity reported as: %s", self._name, activity_name)
         self._current_activity = activity_name
         self._state = bool(self._current_activity != 'PowerOff')
         self.schedule_update_ha_state()
 
+    def new_config(self, _ = None):
+        """Call for updating the current activity."""
+        _LOGGER.debug("%s configuration has been updated", self._name)
+        self.new_activity(self._client.current_activity)
+
+        _, self._current_activity = self._client.current_activity
+        self._state = bool(self._current_activity != 'PowerOff')
+        self.schedule_update_ha_state()
+        self.write_config_file()
```

Comment:
This should be run in the executor. It seems we're mixing things that possibly should be run in the event loop with things that should run in the thread pool.